frontend/src/utils/index.js

```diff
@@ -643,3 +643,26 @@ export function omitKeysWithSuffix (obj, suffix) {
   const keys = Object.keys(obj).filter(key => key.endsWith(suffix))
   return omit(obj, keys)
 }
+
+export function parseNumericalAbbreviation (abbreviatedNumber) {
+  const regex = /(\d+(?:\.\d+)?)(k|M|B)?/i
+  const match = abbreviatedNumber.match(regex)
+
+  if (!match) {
+    return null
+  }
+
+  const number = parseFloat(match[1])
+  const suffix = match[2]
+
+  switch (suffix?.toLowerCase()) {
+    case 'k':
+      return number * 1e3
+    case 'm':
+      return number * 1e6
+    case 'b':
+      return number * 1e9
+    default:
+      return number
+  }
```

Comment:
The regex already makes sure the number has correct format.
```suggestion
  const suffixFactors = { k: 1e3, m: 1e6, b: 1e9, t: 1e12 }
  const factor = suffixFactors [suffix?.toLowerCase()] ?? 1
  return Number(number) * factor
```